compiler/circle-interpreter-test/src/circle-interpreter.test.cpp

```diff
@@ -0,0 +1,236 @@
+/*
+ * Copyright (c) 2022 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <gtest/gtest.h>
+
+#include <cstdlib>
+#include <cstring>
+#include <fstream>
+#include <vector>
+
+class circle_interpreter_test : public ::testing::Test
+{
+protected:
+  bool initialize(void);
+  bool run(const std::string &command);
+
+protected:
+  bool compare(const std::string &file1, const std::string &file2);
+
+protected:
+  std::string _artifacts_path;
+  std::string _circle_interpreter_path;
+  std::string _result;
+};
+
+bool circle_interpreter_test::initialize(void)
+{
+  char *path = std::getenv("ARTIFACTS_PATH");
+  if (path == nullptr)
+  {
+    std::cerr << "ARTIFACTS_PATH not found" << std::endl;
+    return false;
+  }
+  _artifacts_path = path;
+
+  path = std::getenv("CIRCLE_INTERPRETER_PATH");
+  if (path == nullptr)
+  {
+    std::cerr << "CIRCLE_INTERPRETER_PATH  not found" << std::endl;
+    return false;
+  }
+  _circle_interpreter_path = path;
+
+  return true;
+}
+
+bool circle_interpreter_test::run(const std::string &command)
+{
+  std::vector<char> buffer(260);
```

Comment:
What does this `260` mean?

Reply:
Honestly I don't have idea, but I just referred it from `circle-operator.test.cpp`.

Reply:
Good question. 260 came from Windows platform `_MAX_PATH` but not related with this particular usecase. Just assuming one line should be within this length.

Reply:
FYI, [linux kernel](https://github.com/torvalds/linux/blob/v5.11/drivers/tty/n_tty.c#L1681) says maximum line length cannot exceed 4095 characters.

Reply:
Thank you for information. `Note` command or creating a variable named `LINE_MAX_LEN` or something would be helpful.